src/Microsoft.AspNetCore.ResponseCompression/BodyWrapperStream.cs

```diff
@@ -205,13 +214,12 @@ public void DisableRequestBuffering()
         // For this to be effective it needs to be called before the first write.
         public void DisableResponseBuffering()
         {
-            if (!_compressionProvider.SupportsFlush)
+            if (CompressionProvider?.SupportsFlush == false)
```

Comment:
This creates an ordering problem between setting ContentType and calling DisableResponseBuffering. Let's finish our other discussion with @davidfowl around the desired behavior of DisableResponseBuffering before we decide what to do here.